ride_share.rb

```diff
@@ -0,0 +1,237 @@
+          }
+        ],
+      DR0001:
+      [
+        { :date=>"3rd Feb 2016",
+          :cost=>10,
+          :rider_id=>"RD0003",
+          :rating=>3
+          },
+        { :date=>"3rd Feb 2016",
+          :cost=>30,
+          :rider_id=>"RD0015",
+          :rating=>4
+          },
+        { :date=>"5th Feb 2016",
+          :cost=>45,
+          :rider_id=>"RD0003",
+          :rating=>2
+          }
+        ],
+      DR0002:
+      [
+        { :date=>"3rd Feb 2016",
+          :cost=>25,
+          :rider_id=>"RD0073",
+          :rating=>5
+          },
+        { :date=>"4th Feb 2016",
+          :cost=>15,
+          :rider_id=>"RD0013",
+          :rating=>1
+          },
+        { :date=>"5th Feb 2016",
+          :cost=>35,
+          :rider_id=>"RD0066",
+          :rating=>3
+          }
+        ],
+      DR0003:
+      [
+        { :date=>"4th Feb 2016",
+          :cost=>5,
+          :rider_id=>"RD0066",
+          :rating=>5
+          },
+        { :date=>"5th Feb 2016",
+          :cost=>50,
+          :rider_id=>"RD0003",
+          :rating=>2
+          }
+        ]
+    }
+}
+
+########################################################
+# Step 4: Total Driver's Earnings and Number of Rides
+
+# Use an iteration blocks to print the following answers:
+# - the number of rides each driver has given
+# - the total amount of money each driver has made
+# - the average rating for each driver
+# - Which driver made the most money?
+# - Which driver has the highest average rating?
+
+puts "The number of rides each driver has given:"
+ride_share_raw_data[:drivers].each do |driver, rides|
+  if rides.count < 2
+    puts "==> Driver \"#{ driver }\" has given #{ rides.count } ride."
+  else
+    puts "==> Driver \"#{ driver }\" has given #{ rides.count } rides."
+  end
+end
+
+puts "\nThe total amount of money each driver has made:"
+ride_share_raw_data[:drivers].each do |driver, rides|
+  total_amount = rides.sum { |ride_detail| ride_detail[:cost] }
+  puts "==> Driver \"#{ driver }\" has earned $#{ total_amount.to_f }."
+end
+
+puts "\nThe average rating for each driver:"
+ride_share_raw_data[:drivers].each do |driver, rides|
+  total_rating = rides.sum { |ride_detail| ride_detail[:rating] }
+  puts "==> Driver \"#{ driver }\" has got the average rating for #{ (total_rating / rides.count).to_f }."
+end
+
+# Handle ranking questions, even with a tie
+def drivers_ranking(data_for_riders, request_info)
+  if request_info == "cost"
+    money_ranking = Hash.new
+    data_for_riders.each do |driver, rides|
+      money_ranking[driver] = rides.sum { |info| info[:cost] }
+    end
+    ranking_for_drivers = money_ranking.select { |driver, value| value == (money_ranking.max_by { |driver,earning| earning })[1] }
+
+  elsif request_info == "rating"
+    ave_rating_ranking = Hash.new
+    data_for_riders.each do |driver, rides|
+      ave_rating_ranking[driver] = (rides.sum { |info| info[:rating] } / rides.count).to_f
```

Comment:
Converting to a float after doing the division means that the data has already been truncated.

```suggestion
      ave_rating_ranking[driver] = (rides.sum { |info| info[:rating] } / rides.count.to_f)
```

Reply:
OMG! That was very tricky, I didn't notice it! Thank you Kaida <3